Model/review.model.js

```diff
@@ -1,5 +1,38 @@
 const db = require("../db/connection.js");
 
+exports.selectReviews = (query) => {
+  const columnName = Object.keys(query)[0];
+  const whereValue = query[columnName];
+
+  let queryStr =
+    "SELECT reviews.*, COUNT(comment_id) AS comment_count FROM reviews LEFT JOIN comments ON comments.review_id = reviews.review_id";
+  if (columnName) {
+    queryStr += ` WHERE ${columnName} = $1`;
+  }
```

Comment:
I would want a LOT more protection here. This is leaving us immensely wide open for SQL injection. As simple as saying ``'/api/reviews?;DROP TABLE users IF EXISTS;=1'``

To guard against this, we can make an array of acceptable column names/queries and check that the columnName is included in that array before we just template it directly into the query string

Reply:
See I had that then removed it... for no good reason....